### DISCUSSION TITLE
useFetch change from 3.8.2 to 3.9.x throws TypeScript errors

### DISCUSSION THREAD

**Discussion by lmc3s3f0r** [nuxt/nuxt#25267] [Questions] [answered]
This week I updated Nuxt from 3.8.2 to 3.9.3, but had to go back since I wasn't able to figure out a TypeScript error I started getting.

The error seems to come from a change in useFetch, which has passed from returning "globalThis.Ref<T | null>" to "Ref<T | null>".

This resulted in being unable to use "data" returned from useFetch in <template>, getting errors similar to these: 

`"Property 'data' does not exist on type 'Ref<Organization | null>'.ts(2339)"`

`"Type 'Ref<boolean>' is not assignable to type 'boolean'.ts(2322)"`

in a component defined like this:

```
<template>
      <ResourcesPageList
        :data="data?.data"
        :pending="pending"
      />
    </template>
  </PageContainer>
</template>

<script lang="ts" setup>
[...]
const { data, error, pending } = await useFetch<Organization>('/api/getData')
[...]
</script>

```

Not sure if it's a bug or if something has to be changed when using useFetch in 3.9.3, but I found nothing in the docs.

**Comment by happy-turtle** [comment]
I got the same problem. I tried to set up a minimal reproduction, but on a fresh Nuxt project typecheck runs fine on `useFetch`. I get errors on pinia and vueuse composables. Do you have extra dependencies as well? Maybe it's caused by one of those. This prevents me from upgrading Nuxt at the moment.

**Comment by lmc3s3f0r** [comment]
It's a big project, so yeah, there are so many other dependencies. The thing is, how to debug which one is causing that? ^_^U

**Reply by happy-turtle** [reply]
Yes, hard to tell. Will keep an eye out whether I can find anything 🤷

**Comment by lmc3s3f0r** [comment]
Bumping so it doesn't get lost.

**Reply by TheAlexLichter** [reply]
Best to create a reproduction and raise an issue!

**Reply by lmc3s3f0r** [reply]
Agreed. Will do next time :)

**Comment by happy-turtle** [comment]
Still hard to tell, but probably related:

- https://github.com/vuejs/pinia/issues/2086
- https://github.com/vuejs/core/issues/10203
- https://github.com/vuejs/core/commit/ad274737015c36906d76f3189203093fa3a2e4e7
- https://github.com/vuejs/core/issues/6766

**Comment by TheAlexLichter** [comment, marked as answer]
Let's follow along in https://github.com/nuxt/nuxt/issues/25443
